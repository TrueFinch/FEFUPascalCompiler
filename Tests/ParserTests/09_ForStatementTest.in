program a;
var
    i: integer;
begin
    for i := 0 to 5 do
        write(i);
end.