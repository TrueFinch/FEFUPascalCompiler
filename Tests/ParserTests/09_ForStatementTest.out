                  a                    
                  ┼                    
              MainBlock                
       ┌──────────┴───────┐            
      var         CompoundStatement    
       ┼                  ┼            
 SimpleVarDecl      ForStatement       
 ┌─────┴┐       ┌───┬─────┴───┐        
 i  SimpleType  i   to   FunctionCall  
       ┌┴          ┌┴─┐    ┌──┴─┐      
    integer        0  5  write  i      